src/workerd/api/http.c++

```diff
@@ -417,6 +418,143 @@ bool Headers::inspectImmutable() {
   return guard != Guard::NONE;
 }
 
+// -----------------------------------------------------------------------------
+// serialization of headers
+//
+// http-over-capnp.capnp has a nice list of common header names, taken from the HTTP/2 standard.
+// We'll use it as an optimization.
+//
+// Note that using numeric IDs for headers implies we lose the original capitalization. However,
+// the JS Headers API doesn't actually give the application any way to observe the capitalization
+// of header names -- it only becomes relevant when serializing over HTTP/1.1. And at that point,
+// we are actually free to change the capitalization anyway, and we commonly do (KJ itself will
+// normalize capitalization of all registered headers, and http-over-capnp also loses
+// capitalization). So, it's certainly not worth it to try to keep the original capitalization
+// across serialization.
```

Comment:
Probably easier just to note that http defines header names as case-insensitive and ignoring case here is perfectly fine.

Reply:
Well, `Headers` does go out of its way to remember the original capitalization. Though given it's not visible at all in the API, maybe we should change that?

Reply:
I would. The spec is pretty clear on them being case-insensitive https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1

Reply:
Of course. Our `Headers` implementation is case-insensitive, but also case-preserving, kind of like the Windows filesystem. The question is whether we should abandon case preservation.